xbmc/windowing/osx/WinSystemOSX.mm

```diff
@@ -1136,16 +1195,21 @@ static void DisplayReconfigured(CGDirectDisplayID display,
   const NSUInteger dispIdx =
       GetDisplayIndex(CServiceBroker::GetSettingsComponent()->GetSettings()->GetString(
           CSettings::SETTING_VIDEOSCREEN_MONITOR));
-  NSScreen* screen = nil;
-  if (dispIdx < NSScreen.screens.count)
+  NSScreen* currentScreen = nil;
+  NSScreen* targetScreen = nil;
+  if (dispIdx < NSScreen.screens.count && m_lastDisplayNr < NSScreen.screens.count)
   {
-    screen = [NSScreen.screens objectAtIndex:dispIdx];
+    currentScreen = [NSScreen.screens objectAtIndex:m_lastDisplayNr];
+    targetScreen = [NSScreen.screens objectAtIndex:dispIdx];
   }
   // move the window to the center of the new screen
-  if (dispIdx != m_lastDisplayNr && screen)
+  if (dispIdx != m_lastDisplayNr && targetScreen && currentScreen)
   {
-    NSPoint windowPos =
-        NSMakePoint(NSMidX(screen.frame) - m_nWidth / 2, NSMidY(screen.frame) - m_nHeight / 2);
+    NSUInteger backingFactor = currentScreen.backingScaleFactor > targetScreen.backingScaleFactor
+                                   ? currentScreen.backingScaleFactor
+                                   : 1;
```

Comment:
Shouldn't you just take the maximum of scale factors?

Not sure I fully understand this piece of code inside `if` regarding transformation of screen's center. Could you explain?

Reply:
The issue only occurs when you move the window from a hidpi display (e.g. the main one) to a monitor with non-hidpi resolution. In those cases to center the window on the screen you need to divide/compensate the window dimensions by the original backing scale factor (otherwise you move it to a wrong position). 
All other cases should be fine, the window is just a bit smaller if originally on a non-hidpi and moving to hidpi (center is still calculated right). There might be a few edge cases but I believe currently the backing factor is either 2 or simply 1.

Reply:
tldr: `m_width` and `m_height` refer to pixels (not points) after the resize callbacks are invoked - hence if we trigger a move programatically we need to compensate the dimensions on the future display if the backing scale is smaller (resize is also invoked on the future display).

Reply:
now it's clear, thanks!

what if in the future you'll be moving from 3x screen to 2x? that literal `1` looks wrong. Probably just take max of the two values?

Reply:
Yeah as long as we keep the condition (moving from higher dpi to lower dpi) that can make it a bit future proof. Will add

Reply:
Thinking a bit more about this:
- Moving from a higher dpi source screen to a lower dpi target screen -> window dimensions need to be corrected by the ratio between the two
- All other cases (source lower/equal than target) -> window dimensions are contained (or equal) within the new screen so we should use the source ratio